component/src/test/java/org/wso2/extension/siddhi/io/websocket/source/WebSocketSourceTestCase.java

```diff
@@ -0,0 +1,253 @@
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.extension.siddhi.io.websocket.source;
+
+import org.apache.log4j.Logger;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+import org.wso2.extension.siddhi.io.websocket.util.WebSocketServer;
+import org.wso2.siddhi.core.SiddhiAppRuntime;
+import org.wso2.siddhi.core.SiddhiManager;
+import org.wso2.siddhi.core.event.Event;
+import org.wso2.siddhi.core.exception.SiddhiAppRuntimeException;
+import org.wso2.siddhi.core.stream.input.InputHandler;
+import org.wso2.siddhi.core.stream.output.StreamCallback;
+import org.wso2.siddhi.core.util.SiddhiTestHelper;
+import org.wso2.siddhi.query.api.exception.SiddhiAppValidationException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class WebSocketSourceTestCase {
+    private static final Logger log = Logger.getLogger(WebSocketSourceTestCase.class);
+    private AtomicInteger eventCount = new AtomicInteger(0);
+    private AtomicInteger eventCount1 = new AtomicInteger(0);
+    private List<String> receivedEventNameList;
+    private int waitTime = 50;
+    private int timeout = 30000;
+
+    @BeforeMethod
+    public void init() {
+        eventCount.set(0);
+        eventCount1.set(0);
+        WebSocketServer.start();
+    }
+
+    @AfterMethod
+    public void serverClose() throws Exception {
+        WebSocketServer.stop();
+    }
+
+
+    @Test
+    public void websocketSinkAndSourceBinaryMapTestCase() throws InterruptedException {
+        receivedEventNameList = new ArrayList<>(3);
+        SiddhiManager siddhiManager = new SiddhiManager();
+        SiddhiAppRuntime siddhiAppRuntime = siddhiManager
+                .createSiddhiAppRuntime(
+                        "@App:name('TestExecutionPlan') " +
+                                "define stream FooStream1 (symbol string, price float, volume long); " +
+                                "@info(name = 'query1') " +
+                                "@source(type='websocket', url = 'ws://localhost:8025/abc', " +
+                                "@map(type='binary'))" +
+                                "Define stream BarStream1 (symbol string, price float, volume long);" +
+                                "from FooStream1 select symbol, price, volume insert into BarStream1;");
+        siddhiAppRuntime.addCallback("BarStream1", new StreamCallback() {
+            @Override
+            public void receive(Event[] events) {
+                for (Event event : events) {
+                    log.info(event);
```

Comment:
What is the purpose of this log? Who is going to read it?

Apply for other places as well.

Reply:
fixed in 9fea7d3